spec/spec_helper.rb

```diff
@@ -48,4 +52,45 @@ def with_memory_logger level = nil
       Asciidoctor::LoggerManager.logger = old_logger
     end
   end
+
+  def compare_images reference, actual, difference = nil
+    diff = []
+    images = [ChunkyPNG::Image.from_file(reference), ChunkyPNG::Image.from_file(actual)]
+
+    images.first.height.times do |y|
+      images.first.row(y).each_with_index do |pixel, x|
+        diff << [x,y] unless pixel == images.last[x,y]
+      end
+    end
+
+    if diff.length > 0 and difference
+      x, y = diff.map{ |xy| xy[0] }, diff.map{ |xy| xy[1] }
+      images.last.rect(x.min, y.min, x.max, y.max, ChunkyPNG::Color.rgb(0,255,0))
+      images.last.save(difference)
+    end
+
+    diff.length
+  end
+
+end
+
+RSpec::Matchers.define(:match_fixture) do |reference_filename|
+  match do |actual_path|
+    reference_path = File.join __dir__, "integration", "fixtures", testname, reference_filename
+    images_output_dir = output_file File.join "integration", testname
+    FileUtils.mkdir_p images_output_dir
+    output_basename = File.join images_output_dir, File.basename(actual_path, '.pdf')
+    %x( pdftocairo -png "#{actual_path}" "#{output_basename}-actual" )
+    %x( pdftocairo -png "#{reference_path}" "#{output_basename}-reference" )
+
+    # find the generated pairs to compare
+    pixelcount = 0
+    Dir["#{output_basename}-reference-*.png"].each do |reference_page_filename|
```

Comment:
The convention in Asciidoctor is to define interpolated strings using `%()` instead of `""`.